### PR TITLE
Fix parametrized tests

### LINKED ISSUE #809: Fix parameterized testing
Google has updated orchestrator to support parameterized tests. If this works, then we should update the readme with instructions for end users that they should update the androidx test dependency to run parameterized tests.

- https://github.com/android/android-test/releases/tag/androidx-test-1.3.0-beta02
- https://github.com/android/android-test/issues/215

### PR DESCRIPTION
**Pull request by adamfilipow92**
Fixes #809

When test will executed from Flank should have results like same test executed directly on testlab.

## Test directly executed from test lab
https://console.firebase.google.com/u/0/project/flank-open-source/testlab/histories/bh.da0c237aaa33732/matrices/8109159837961280691

## Failed test executed from Flank
https://console.firebase.google.com/u/0/project/flank-open-source/testlab/histories/bh.da0c237aaa33732/matrices/8167982987416424556/executions/bs.38eda8775a39b098
and other with video
https://console.firebase.google.com/u/0/project/flank-open-source/testlab/histories/bh.da0c237aaa33732/matrices/7496090094871625048/executions/bs.160a4ed4fdc86abe

## Checklist

- [x] Documented
- [x] Unit tested
- [x] release_notes.md updated
- [X] Test case with app, yml uploaded
- [x] Flank should correctly run parametrized tests 

## Objectives
Parametrized tests classes should be run without splitting into single methods to avoid run errors.

## Tests status
https://github.com/adamfilipow92/flank_parametrized_tests/tree/master/test_apk

### CODE REVIEW

File: test_runner/src/test/kotlin/ftl/fixtures/test_app_cases/flank-single-parametrized.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+gcloud:
+  app: ./src/test/kotlin/ftl/fixtures/tmp/apk/app-single-success-parametrized-debug.apk
```
Comment (bootstraponline) [inline test_runner/src/test/kotlin/ftl/fixtures/test_app_cases/flank-single-parametrized.yml:2]: Apks shouldn't be added directly to the flank repo, instead we keep them in https://github.com/Flank/test_artifacts to keep the flank repo size small

For the sample app, let's update the source code in this repo so we can easily rebuild the apks. As Android evolves, we'll want to rebuild apks with different options and target platforms.

File: test_runner/src/main/kotlin/ftl/run/platform/android/CreateAndroidTestContext.kt [hunk 2]
```diff
@@ -50,13 +51,34 @@ private fun InstrumentationTestContext.calculateShards(
...
+        file.classDefs.forEach { classDef ->
+            val directory = file.getAnnotationsDirectory(classDef)
+            val className = file.formatClassName(classDef).dropLast(1)
+            val classAnnotations = file.getClassAnnotationValues(directory).map { it.name }
+            if (classAnnotations.any { it.toLowerCase().contains("RunWith".toLowerCase()) } && // sprawdzamy czy klasa ma adnotacje RunWith
+                file.getClassAnnotationValues(directory)
+                    .flatMap { annotations -> annotations.values.values }
+                    .filterIsInstance<DecodedValue.DecodedType>()
+                    .map { it.value }
+                    .any { it.toLowerCase().contains("Parameterized".toLowerCase()) }) {  // sprawdzamy czy klasa zadana jako parametr  adnotacji to Parameterized
```
Comment (jan-gogo) [inline test_runner/src/main/kotlin/ftl/run/platform/android/CreateAndroidTestContext.kt:66]: I wonder if there are any other ways to define tests that can be identified by `AndroidJUnitRunner` as parameterized

File: test_app/app/src/androidTestMultiple/java/com.example.test_app/FajneTesty123.kt [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+            arrayOf(666, "666") ,
+            arrayOf(54321, "54321")
+        )
+    }
+
+    @Test
+    fun shouldHopefullyPass() {
+        assertTrue(paramOne == paramTwo.toInt())
+    }
+}
```
Comment (jan-gogo) [inline test_app/app/src/androidTestMultiple/java/com.example.test_app/FajneTesty123.kt:26]: missing newline

File: test_runner/src/test/kotlin/ftl/fixtures/test_app_cases/flank-single-parametrized.yml (hunk 1 above)
Comment (adamfilipow92) [reply test_runner/src/test/kotlin/ftl/fixtures/test_app_cases/flank-single-parametrized.yml:2]: Changed

File: test_runner/src/main/kotlin/ftl/run/platform/android/CreateAndroidTestContext.kt [hunk 4]
```diff
@@ -51,12 +59,43 @@ private fun InstrumentationTestContext.calculateShards(
...
+
+private fun DexFile.isParametrizedClass(classDef: ClassDefItem): Boolean =
+    getClassAnnotationValues(getAnnotationsDirectory(classDef)).let { annotations: List<TestAnnotation> ->
+        annotations.any { it.name.contains("RunWith", ignoreCase = true) } && annotations
+            .flatMap { it.values.values }
+            .filterIsInstance<DecodedValue.DecodedType>()
+            .any { it.isParameterizedAnnotation() }
+    }
+
+private fun DecodedValue.DecodedType.isParameterizedAnnotation(): Boolean =
```
Comment (bootstraponline) [inline test_runner/src/main/kotlin/ftl/run/platform/android/CreateAndroidTestContext.kt:93]: This is brilliant! ⭐

**Pull request merged**